spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -745,6 +807,20 @@ private void cleanUpAfterCopy(File fileToWriteTo, File resultFile, File original
 		if (this.deleteSourceFiles && originalFile != null) {
 			originalFile.delete();
 		}
+
+		setPermissions(resultFile);
+	}
+
+	/**
+	 * Set permissions on newly written files.
+	 * @param resultFile the file.
+	 * @throws IOException any exception.
+	 * @since 5.0
+	 */
+	protected void setPermissions(File resultFile) throws IOException {
+		if (this.permissions != null) {
+			Files.setPosixFilePermissions(resultFile.toPath(), this.permissions);
```

Comment:
Yes; it does fail; I tested on Windows - comment out the if statements in the test 😄